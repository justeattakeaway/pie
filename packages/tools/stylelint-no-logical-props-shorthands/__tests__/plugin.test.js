import {
    describe,
    it,
    expect,
    beforeEach,
} from 'vitest';

const stylelint = require('stylelint');

const config = {
    extends: 'stylelint-config-standard-scss',
    plugins: [
        '@justeattakeaway/stylelint-no-logical-props-shorthands',
    ],
    rules: {
        '@justeattakeaway/stylelint-no-logical-props-shorthands': true,
    },
};

describe('stylelint-no-logical-props-shorthands', () => {
    let result;

    describe('when given valid CSS', () => {
        const validCSS = `a { margin-block-start: 10px; }
`;
        beforeEach(() => {
            result = stylelint.lint({
                code: validCSS,
                config,
            });
        });

        it('doesn\'t raises an error', () => result.then((data) => {
            expect(data.errored).toBeFalsy();
        }));
    });

    describe('when given invalid CSS', () => {
        const invalidCSS = `a { margin-block: var(--dt-spacing-e); }
`;
        beforeEach(() => {
            result = stylelint.lint({
                code: invalidCSS,
                config,
            });
        });

        it('raises an error', () => result.then((data) => {
            expect(data.errored).toBeTruthy();
        }));

        describe('when autofix is true', () => {
            beforeEach(() => {
                result = stylelint.lint({
                    code: invalidCSS,
                    config,
                    fix: true,
                });
            });

            describe('when a single value is given', () => {
                it('provides the expected suggestion', async () => {
                    const code = `a { margin-block: 10px; }
`;
                    const result = await stylelint.lint({
                        code,
                        config,
                        fix: true,
                    });

                    expect(result.output).toMatchSnapshot();
                });
            });

            describe('when multiple values are given', () => {
                describe('and value does not contains a CSS variable', () => {
                    it('should keep it unchanged', async () => {
                        const code = `a { margin-block: 12px 34px; }
    `;
                        const result = await stylelint.lint({
                            code,
                            config,
                            fix: true,
                        });

                        expect(result.output).toMatchSnapshot();
                    });
                });

                describe('and value contains a CSS variable', () => {
                    it('should provide the expected suggestion', async () => {
                        const code = `a { margin-block: 12px var(--dt-spacing-e); }
    `;
                        const result = await stylelint.lint({
                            code,
                            config,
                            fix: true,
                        });

                        expect(result.output).toMatchSnapshot();
                    });
                });

                describe('and values are complex', () => {
                    it('provides the expected suggestion', async () => {
                        const code = `a { padding-inline: calc(var(--dt-spacing-d) + $nav-item-inline-start-margin) var(--dt-spacing-d); }
    `;
                        const result = await stylelint.lint({
                            code,
                            config,
                            fix: true,
                        });

                        expect(result.output).toMatchSnapshot();
                    });
                });
            });
        });
    });
});
