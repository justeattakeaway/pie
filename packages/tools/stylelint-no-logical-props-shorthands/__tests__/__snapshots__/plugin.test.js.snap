// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`stylelint-no-logical-props-shorthands > when given invalid CSS > when autofix is true > when a single value is given > provides the expected suggestion 1`] = `
"a { margin-block-start: 10px; margin-block-end: 10px; }
"
`;

exports[`stylelint-no-logical-props-shorthands > when given invalid CSS > when autofix is true > when multiple values are given > and values are complex > provides the expected suggestion 1`] = `
"a { padding-inline-start: calc(var(--dt-spacing-d) + $nav-item-inline-start-margin); padding-inline-end: var(--dt-spacing-d); }
"
`;

exports[`stylelint-no-logical-props-shorthands > when given invalid CSS > when autofix is true > when multiple values are given > provides the expected suggestion 1`] = `
"a { margin-block-start: 12px; margin-block-end: 34px; }
"
`;
